src/storage-blob-preview/azext_storage_blob_preview/tests/latest/test_storage_blob_live_scenarios.py

```diff
@@ -87,6 +87,41 @@ def verify_blob_upload_and_download(self, group, account, file_size_kb, blob_typ
             self.assertEqual(file_size_kb * 1024, os.stat(downloaded).st_size,
                              'The download file size is not right.')
 
+    @ResourceGroupPreparer()
+    @StorageAccountPreparer()
+    def test_storage_blob_upload(self, resource_group, storage_account):
+        from azure.cli.core.azclierror import AzureResponseError
+        file_size_kb = 128
+        local_file = self.create_temp_file(file_size_kb, full_random=True)
+
+        container = self.create_random_name(prefix='cont', length=10)
+        blob_name = self.create_random_name(prefix='blob', length=24)
+        account_key = self.cmd('storage account keys list -n {} -g {} --query "[0].value" -otsv'
+                               .format(storage_account, resource_group)).output
+
+        self.set_env('AZURE_STORAGE_ACCOUNT', storage_account)
+        self.set_env('AZURE_STORAGE_KEY', account_key)
+
+        self.cmd('storage container create -n {}'.format(container))
+        # test upload through file path
+        self.cmd('storage blob upload -c {} -f "{}" -n {}'.format(container, local_file, blob_name))
+
+        self.cmd('storage blob exists -n {} -c {}'.format(blob_name, container), checks=JMESPathCheck('exists', True))
+
+        self.cmd('storage blob upload -c {} -f "{}" -n {}'.format(container, local_file, blob_name),
+                 checks=JMESPathCheck('exists', False))
```

Comment:
Why re-upload here? Is this check for `exists` instead of `upload`?

Reply:
removed